tizen/src/extractor.h

```diff
@@ -0,0 +1,145 @@
+#ifndef FLUTTER_PLUGIN_WORKMANAGER_EXTRACTOR_H_
+#define FLUTTER_PLUGIN_WORKMANAGER_EXTRACTOR_H_
+
+#include <flutter/encodable_value.h>
+#include <flutter/method_call.h>
+
+#include <string>
+
+#include "constants.h"
+#include "options.h"
+#include "tasks.h"
+#include "utils.h"
+
+typedef flutter::MethodCall<flutter::EncodableValue> FlMethodCall;
+
+ExistingWorkPolicy ExtractExistingWorkPolicyFromCall(const FlMethodCall &call) {
+    std::string value;
+    const auto &args = *call.arguments();
+    flutter::EncodableMap map = std::get<flutter::EncodableMap>(args);
+    bool succeed = GetValueFromEncodableMap<std::string>(
+        &map, constants::keys::kExistingWorkpolicykey, value);
+    if (succeed) {
+        Upper(value);
+
+        if (value == constants::values::kReplace) {
+            return ExistingWorkPolicy::kReplace;
+        } else if (value == constants::values::kKeep) {
+            return ExistingWorkPolicy::kKeep;
+        } else if (value == constants::values::kAppend) {
+            return ExistingWorkPolicy::kAppend;
+        } else if (value == constants::values::kUpdate) {  // TODO : check
+                                                           // real value
+            return ExistingWorkPolicy::kUpdate;
+        }
+    }
+    return ExistingWorkPolicy::kKeep;
+}
+
+std::optional<BackoffPolicyTaskConfig> ExtractBackoffPolicyConfigFromCall(
+    const FlMethodCall &call, TaskType task_type) {
+    const auto &args = *call.arguments();
+    flutter::EncodableMap map = std::get<flutter::EncodableMap>(args);
+
+    std::string value;
+    if (!GetValueFromEncodableMap(&map, constants::keys::kBackOffPolicyTypeKey,
+                                  value)) {
+        return std::nullopt;
+    }
+
+    BackoffPolicy backoff_policy;
+
+    if (value == constants::values::kExponential) {
+        backoff_policy = BackoffPolicy::kExponential;
+    } else {
+        backoff_policy = BackoffPolicy::kLinear;
+    }
```

Comment:
How about using this type of [function](https://github.com/flutter-tizen/plugins/blob/c23e61c21935a9204fb7169b6892ec26ab1f1fd5/packages/audioplayers/tizen/src/audioplayers_tizen_plugin.cc#L48)?

Reply:
fixed it